uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-notification.component.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-env amd */
+
+define([
+  'leave-absences/shared/modules/components'
+], function (components) {
+  components.component('leaveNotification', {
+    bindings: {
+      eventName: '<'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-notification.html';
+    }],
+    controllerAs: 'notification',
+    controller: ['$log', '$rootScope', 'pubSub', 'LeaveRequest', controller]
+  });
+
+  function controller ($log, $rootScope, pubSub, LeaveRequest) {
+    $log.debug('Component: leave-notification');
+
+    var filters = {};
+    var vm = this;
+    vm.count = 0;
+
+    (function init () {
+      initializeListeners();
+    })();
+
+    /**
+     * Fetch count of leave requests which matches the filter
+     *
+     * @return {Promise}
+     */
+    function fetchCount () {
+      return LeaveRequest.all(filters)
+        .then(function (leaveRequests) {
+          vm.count = leaveRequests.list.length;
+        });
+    }
+
+    /**
+     * Initializes the filter and fetches fetches filtered leave requests
+     * @param {Object} e - Event object
```

Comment:
you are not using this, use `__` for unused parameters, and don't list them in the documentation